File: java-implementation/src/main/java/com/inventory/products/stationary/Notepads.java
package com.inventory.products.stationary;

// Represents a concrete subclass of Stationary
public class Notepads extends Stationary {
    private int pages;

    //  Constructs an instance with specified details with base class attributes included
    public Notepads(int id, String name, double price, int volume, String material, int pages, String supplier) {
        super(id, name, price, volume, material, supplier);
        this.pages = pages;
    }

    // Calculates the discount based on volume and month
    @Override
    public double calculateDiscount(int volume, int month) {
        if (month == 12 && volume > 5) {
            return price * volume * 0.25;
        }
        return 0;
    }

    // Provides a string representation the details
    @Override
    public String displayProductInfo() {
        return "ID: " + id + "\nBrand Name: " + name + "\nPaper Type: " + material
               + "\nNo. of Pages: " + pages + "\nPrice: €" + price
               + "\nVolume: " + volume + " units\nSupplier: " + supplier + "\n\n";
    }
}

File: cpp-implementation/src/products/Stationary/Notepads.cpp
#include "Product.ih"

// Constructor implementation for Notepads
Notepads::Notepads(int id, string name, double price, int volume, 
                    string material, int pages, string supplier)
    : Stationary(id, name, price, volume, material, supplier), pages(pages) {}

// Calculates the discount depending on the current month and the packaged quantity
double Notepads::calculateDiscount(int volume, int month) const {
    if (month == 12 && volume > 5) {
        return price * volume * 0.25;
    }
    return 0;
}

// Displays all the information 
string Notepads::displayProductInfo() const {
    ostringstream info;
    info << "ID: " << id << "\nNotepad Name: " << name << "\nSupplier: " << getSupplier()
         << "\nMaterial: " << material << "\nNumber of Pages: " << pages
         << "\nPrice: €" << price << "\nVolume: " << volume << " units\n\n";
    return info.str();
}
